packages/openai/src/openai-chat-language-model.ts

```diff
@@ -283,15 +95,21 @@ export class OpenAIChatLanguageModel implements LanguageModelV1 {
     const { messages: rawPrompt, ...rawSettings } = args;
     const choice = response.choices[0];
 
-    const providerMetadata =
-      response.usage?.completion_tokens_details?.reasoning_tokens != null
-        ? {
-            openai: {
-              reasoningTokens:
-                response.usage?.completion_tokens_details?.reasoning_tokens,
-            },
-          }
-        : undefined;
+    let providerMetadata: LanguageModelV1ProviderMetadata | undefined;
+    if (
+      response.usage?.completion_tokens_details?.reasoning_tokens !== null ||
+      response.usage?.prompt_tokens_details?.cached_tokens !== null
+    ) {
+      providerMetadata = { openai: {} };
+      if (response.usage?.completion_tokens_details?.reasoning_tokens) {
+        providerMetadata.openai.reasoningTokens =
+          response.usage?.completion_tokens_details?.reasoning_tokens;
+      }
+      if (response.usage?.prompt_tokens_details?.cached_tokens) {
+        providerMetadata.openai.cachedPromptTokens =
+          response.usage?.prompt_tokens_details?.cached_tokens;
+      }
+    }
```

Comment:
also needs support in doStream

Reply:
(reasoning tokens don't need doStream support because o1 does not support streaming)

Reply:
ahhh got it.

Reply:
not sure where to look for this - can't find anything about provider metadata in `doStream` for non-reasoning-related models. will dig in tomorrow.

Reply:
added streaming support, example, and test @lgrammel :)